legal-api/src/legal_api/services/business_details_version.py

```diff
@@ -212,102 +219,98 @@ def get_company_details_revision(filing_id, legal_entity_id) -> dict:
+            business_version = (
+                history_cls(LegalEntity) if business_revision.is_legal_entity else history_cls(AlternateName)
+            )
 
-        return le_revision if le_revision else business
+            business_revision = (
+                db.session.query(business_version)
+                .filter(business_version.change_filing_id == filing.id)
+                .filter(business_version.id == identifier)
+                .first()
+            )
+        return business_revision
 
     @staticmethod
     def find_last_value_from_business_revision(
-        filing, legal_entity, is_dissolution_date=False, is_restoration_expiry_date=False
-    ) -> LegalEntity:
+        filing, business, is_dissolution_date=False, is_restoration_expiry_date=False
+    ) -> any:
         """Get business info with last value of dissolution_date or restoration_expiry_date."""
         conditions = []
 
-        legal_entity_version = history_cls(LegalEntity)
+        business_version = history_cls(LegalEntity) if business.is_legal_entity else history_cls(AlternateName)
+
         if is_dissolution_date:
-            if legal_entity.dissolution_date:
-                return legal_entity
+            if business.dissolution_date:
+                return business
             conditions.append(
-                legal_entity_version.dissolution_date != None  # noqa: E711,E501;
+                business_version.dissolution_date != None  # noqa: E711,E501;
             )  # pylint: disable=singleton-comparison
         elif is_restoration_expiry_date:
-            if legal_entity.restoration_expiry_date:
-                return legal_entity
+            if business.restoration_expiry_date:
+                return business
             conditions.append(
-                legal_entity_version.restoration_expiry_date != None  # noqa: E711,E501;
+                business_version.restoration_expiry_date != None  # noqa: E711,E501;
             )  # pylint: disable=singleton-comparison
 
-        le_revision = (
-            db.session.query(legal_entity_version)
-            .filter(legal_entity_version.change_filing_id < filing.id)
-            .filter(legal_entity_version.id == legal_entity.id)
+        business_revision = (
+            db.session.query(business_version)
+            .filter(business_version.change_filing_id < filing.id)
+            .filter(business_version.id == business.id)
             .filter(*conditions)
-            .order_by(legal_entity_version.change_filing_id)
+            .order_by(business_version.change_filing_id)
             .first()
         )
-        return le_revision
+        return business_revision
 
     @staticmethod
-    def get_office_revision(filing, legal_entity_id) -> dict:  # pylint: disable=too-many-locals
+    def get_office_revision(filing, legal_entity_id, alternate_name_id=None) -> dict:  # pylint: disable=too-many-locals
```

Comment:
can we just pass in the business object and just remove `legal_entity_id` and `alternate_name_id`?

Reply:
Will update both `get_business_revision_obj` and `get_office_revision` to pass in business object